### PR TITLE
Issue #PS-313 feat: Edit User Profile

### PR DESCRIPTION
**Pull request by upendraTekdi**
## Description

Edit User Profile

### Changes

List the changes done to fix a bug or introducing a new feature.

## How to test

Describe the steps required to test the changes proposed in the pull request.

Please consider using the closing keyword if the pull request is proposed to
fix an issue already created in the repository
(https://help.github.com/articles/closing-issues-using-keywords/)

## Summary by CodeRabbit

- **New Features**
  - Updated profile redirection to Teacher Profile page based on the pathname.
  - Enhanced Teacher Profile page with image upload, gender selection, and improved field handling.

- **Localization**
  - Expanded language support for "View Full Profile," "Close," and "Learner Details" in English, Hindi, and Marathi.

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -68,8 +68,8 @@ const Header: React.FC = () => {
   }));
 
   const handleProfileClick = () => {
-    if (pathname !== '/my-profile') {
-      router.push('/my-profile');
+    if (pathname !== '/teacherProfile') {
+      router.push('/teacherProfile');
```
Comment (itsvick) [inline src/components/Header.tsx:72]: @upendraTekdi avoid using camelCase in the route

File: src/pages/teacherProfile.tsx [hunk 2]
```diff
@@ -522,35 +389,38 @@ const MyProfile = () => {
...
                 borderRadius={'12px'}
                 border={'1px'}
                 bgcolor="warning.A400"
                 display="flex"
                 flexDirection="column"
               >
-                <Image
-                  src={prathamProfileUrl}
+                <img
+                  src={image}
```
Comment (itsvick) [inline src/pages/teacherProfile.tsx:405]: @upendraTekdi use Image component from the Next

File: src/components/Header.tsx (hunk 1 above)
Comment (upendraTekdi) [reply src/components/Header.tsx:72]: okay

File: src/pages/teacherProfile.tsx (hunk 2 above)
Comment (upendraTekdi) [reply src/pages/teacherProfile.tsx]: okay

**Pull request merged**